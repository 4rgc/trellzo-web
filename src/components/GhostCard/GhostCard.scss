@use '../Card/card-variables' as v;

$border-size: 4px;

.ghost-card.card {
	border: $border-size grey dashed;
	border-radius: 20px;

	&:hover {
		background-color: rgba($color: lightgrey, $alpha: 0.9);
	}

	&.card-sm {
		height: calc(map-get($map: v.$sizes, $key: 'sm') - ($border-size * 2));
	}

	&.card-md {
		height: calc(map-get($map: v.$sizes, $key: 'md') - ($border-size * 2));
	}

	&.card-lg {
		height: calc(map-get($map: v.$sizes, $key: 'lg') - ($border-size * 2));
	}
}

.ghost-card-text-container {
	text-align: center;
}

.ghost-card-text {
	color: grey;
}
