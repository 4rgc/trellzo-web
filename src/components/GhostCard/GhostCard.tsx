import Card from '../Card';
import { ICardProps } from '../Card/Card';
import './GhostCard.scss';

export interface IGhostCardProps extends ICardProps {
	innerText?: string;
}

const GhostCard: React.FC<IGhostCardProps> = (props) => {
	const { innerText, ...otherProps } = props;

	let forceDisableImage = false;
	if (otherProps.size === 'sm') {
		forceDisableImage = true;
	}

	return (
		<Card
			isImageDisabled={forceDisableImage}
			className="ghost-card"
			{...otherProps}
		>
			<div className="ghost-card-text-container">
				<b className="ghost-card-text">{innerText}</b>
			</div>
		</Card>
	);
};

export default GhostCard;
