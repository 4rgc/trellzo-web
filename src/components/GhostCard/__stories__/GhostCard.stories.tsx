import { StoryFn, Meta } from '@storybook/react';

import GhostCard from '../GhostCard';

export default {
	title: 'Components/GhostCard',
	component: GhostCard,
	args: {
		innerText: 'Add a new asdf...',
	},
	argTypes: {
		onClick: { action: 'clicked' },
	},
} as Meta<typeof GhostCard>;

const Template: StoryFn<typeof GhostCard> = ({ children, ...args }) => (
	<GhostCard {...args} style={{ width: '200px' }}>
		{children}
	</GhostCard>
);

export const Default = Template.bind({});

export const NoImage = Template.bind({});
NoImage.args = { isImageDisabled: true };

export const Small = Template.bind({});
Small.args = { size: 'sm' };

export const Large = Template.bind({});
Large.args = { size: 'lg' };
