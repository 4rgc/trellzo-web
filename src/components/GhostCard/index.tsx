import GhostCard from './GhostCard';

export default GhostCard;
