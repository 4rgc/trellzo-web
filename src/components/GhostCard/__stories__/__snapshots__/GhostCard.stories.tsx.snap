// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/GhostCard Default smoke-test 1`] = `
<div class="card card-md card-clickable ghost-card"
     style="width: 200px;"
>
  <svg viewbox="0 0 500 500"
       xmlns="http://www.w3.org/2000/svg"
       fill="black"
       stroke="black"
       class="card-cover"
       preserveaspectratio="xMinYMin slice"
  >
    <g transform="matrix(1, 0, 0, 1, 146.358994, 113.460999)"
       paint-order="fill markers stroke"
    >
      <path d="m118.507 133.514-34.249 34.249-15.968-15.968-41.938 41.937H178.726z"
            opacity="0.675"
            fill="#fff"
      >
      </path>
      <circle cx="58.217"
              cy="108.555"
              r="11.773"
              opacity="0.675"
              fill="#fff"
      >
      </circle>
      <path fill="none"
            d="M26.111 77.634h152.614v116.099H26.111z"
      >
      </path>
    </g>
  </svg>
  <div class="card-divider">
  </div>
  <div class="card-container-no-text">
    <div class="ghost-card-text-container">
      <b class="ghost-card-text">
        Add a new asdf...
      </b>
    </div>
  </div>
</div>
`;

exports[`Components/GhostCard Large smoke-test 1`] = `
<div class="card card-lg card-clickable ghost-card"
     style="width: 200px;"
>
  <svg viewbox="0 0 500 500"
       xmlns="http://www.w3.org/2000/svg"
       fill="black"
       stroke="black"
       class="card-cover"
       preserveaspectratio="xMinYMin slice"
  >
    <g transform="matrix(1, 0, 0, 1, 146.358994, 113.460999)"
       paint-order="fill markers stroke"
    >
      <path d="m118.507 133.514-34.249 34.249-15.968-15.968-41.938 41.937H178.726z"
            opacity="0.675"
            fill="#fff"
      >
      </path>
      <circle cx="58.217"
              cy="108.555"
              r="11.773"
              opacity="0.675"
              fill="#fff"
      >
      </circle>
      <path fill="none"
            d="M26.111 77.634h152.614v116.099H26.111z"
      >
      </path>
    </g>
  </svg>
  <div class="card-divider">
  </div>
  <div class="card-container-no-text">
    <div class="ghost-card-text-container">
      <b class="ghost-card-text">
        Add a new asdf...
      </b>
    </div>
  </div>
</div>
`;

exports[`Components/GhostCard NoImage smoke-test 1`] = `
<div class="card card-md card-clickable card-noimage ghost-card"
     style="width: 200px;"
>
  <div class="card-container-no-text">
    <div class="ghost-card-text-container">
      <b class="ghost-card-text">
        Add a new asdf...
      </b>
    </div>
  </div>
</div>
`;

exports[`Components/GhostCard Small smoke-test 1`] = `
<div class="card card-sm card-clickable card-noimage ghost-card"
     style="width: 200px;"
>
  <div class="card-container-no-text">
    <div class="ghost-card-text-container">
      <b class="ghost-card-text">
        Add a new asdf...
      </b>
    </div>
  </div>
</div>
`;
